### PR TITLE
Fix #7976 Pulldown menus

### LINKED ISSUE #7976: FXIOS-1485 ⁃ Switch to system-native Pull-down Menu
**Vision**

As a user, I want to see more components that are familiar to me to make it easier for me to find what I am looking for and get what I need to do done.

**Acceptance Criteria**

* Use the system-native Pulldown menu for:
Application menu [Figma](https://www.figma.com/file/lfgj15zbNkL9Pp4hdZGJgp/Proton-Mobile-UI?node-id=339%3A27332)
Page Actions menu [Figma](https://www.figma.com/file/lfgj15zbNkL9Pp4hdZGJgp/Proton-Mobile-UI?node-id=422%3A1)
* on iPhone _and_ iPad
* for dark and light mode
* ensure accessibility is not broken
* toggles are replaced with text and icon
Instead of a toggle changing when the menu item is tapped, the icon and text will change.

ex. if you click "Block Images" menu item, the text changes to "Show Images" and the icon changes.

App menu:
                -Turn on Night Mode / Turn off Night Mode (use existing moon icon for both 
                -Block Images / Show Images (same icon for block, new icon for show) [Figma](https://www.figma.com/file/lfgj15zbNkL9Pp4hdZGJgp/Proton-Mobile-UI?node-id=339%3A27332)

Page actions menu:
                 -Request Desktop Site / Request Mobile Site (desktop icon, mobile icon) [Figma](https://www.figma.com/file/lfgj15zbNkL9Pp4hdZGJgp/Proton-Mobile-UI?node-id=422%3A1)

Note: this includes moving the icons to the right of the menu item

Note: For iOS < 14 where it is not supported, fallback to Action Sheets (iOS) and Popovers (iPadOS).

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1485)

### PR DESCRIPTION
**Pull request by dnarcese**
- applies to app menu and page actions menu

### CODE REVIEW

**Comment by nbhasin2** [comment]
I am not sure if this is happening on `main` but if you are in `Private mode` and try to add website to shortcuts then it isn't behaving correctly

https://user-images.githubusercontent.com/8919439/115765285-8fdb9b80-a374-11eb-8d99-67e223db8b7e.mov

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+TabToolbarDelegate.swift [hunk 1]
```diff
@@ -99,24 +99,23 @@ extension BrowserViewController: TabToolbarDelegate, PhotonActionSheetProtocol {
                 TelemetryWrapper.recordEvent(category: .action, method: .tap, object: .logins)
             }
         }
-
-        let optionalActions = [syncAction, viewLogins].compactMap { $0 }
+        
+        var section1 = getLibraryActions(vcDelegate: self)
+        var section2 = getOtherPanelActions(vcDelegate: self)
+        let section3 = getSettingsAction(vcDelegate: self)
```
Comment (nbhasin2) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+TabToolbarDelegate.swift:105]: nit: Any reason why section1 and section2 are vars and not let like sections3

Comment (dnarcese) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+TabToolbarDelegate.swift:105]: I am appending to them later

**Comment by dnarcese** [comment]
re the private mode mode... I can't reproduce. If you open a new private tab does it still happen? Wondering if its broken on existing tabs?

**Comment by nbhasin2** [comment]
> re the private mode bug... I can't reproduce. If you open a new private tab does it still happen? Wondering if its broken on existing tabs?


https://user-images.githubusercontent.com/8919439/115770272-7c333380-a37a-11eb-97a8-f029523cf34c.mov

@dnarcese this is what I am doing is that the same you did? I opened a new tab and went to that website.

**Comment by dnarcese** [comment]
@nbhasin2 ya. Only difference is I am using chron tabs

https://user-images.githubusercontent.com/11432165/115770858-3a56bd00-a37b-11eb-9248-34441ee51b82.MP4

**Comment by nbhasin2** [comment]
@dnarcese it works in Chron Tabs and is broken for regular tabs :(

**Comment by nbhasin2** [comment]
image

The `X` seems different size than others, something to maybe look at

**Comment by dnarcese** [comment]
> image
> 
> The `X` seems different size than others, something to maybe look at

I didn't change this. This should be an existing issue

File: Client/Frontend/Widgets/PhotonActionSheet/PhotonActionSheet.swift [hunk 2]
```diff
@@ -274,7 +276,7 @@ class PhotonActionSheet: UIViewController, UITableViewDelegate, UITableViewDataS
             self.dismiss(nil)
             return
         }
-
+
```
Comment (nbhasin2) [inline Client/Frontend/Widgets/PhotonActionSheet/PhotonActionSheet.swift:279]: nit: remove spacing
```suggestion

```

File: Client/Frontend/Widgets/PhotonActionSheet/PhotonActionSheet.swift [hunk 3]
```diff
@@ -298,47 +301,66 @@ class PhotonActionSheet: UIViewController, UITableViewDelegate, UITableViewDataS
...
+            if site != nil || title != nil {
+                return PhotonActionSheetUX.SeparatorRowHeight
             }
-            return 6
         }
 
         return PhotonActionSheetUX.SeparatorRowHeight
     }
-
+
```
Comment (nbhasin2) [inline Client/Frontend/Widgets/PhotonActionSheet/PhotonActionSheet.swift:343]: nit: spacing
```suggestion

```

**Review by nbhasin2 — APPROVED** [review]

**Pull request merged**